### ISSUE TITLE
Two similar but different geometries giving exact same results

### ISSUE THREAD

**Issue by MortenHofft** [gbif/occurrence#38]
[POLYGON ((-180 60.444, -180 60.462, 180 60.462, 180 60.444, -180 60.444))](http://api.gbif.org/v1/occurrence/search?has_coordinate=true&has_geospatial_issue=false&geometry=POLYGON%20((-180%2060.444,%20-180%2060.462,%20180%2060.462,%20180%2060.444,%20-180%2060.444)))

[POLYGON ((-180 60.426, -180 60.444, 180 60.444, 180 60.426, -180 60.426))](http://api.gbif.org/v1/occurrence/search?has_coordinate=true&has_geospatial_issue=false&geometry=POLYGON%20((-180%2060.426,%20-180%2060.444,%20180%2060.444,%20180%2060.426,%20-180%2060.426)))

both return 37105724 results. that seems unlikely.

same goes for polygons less wide
https://www.gbif.org/occurrence/search?has_coordinate=true&has_geospatial_issue=false&geometry=POLYGON%20((-30%2060.444,%20-30%2060.462,%2030%2060.462,%2030%2060.444,%20-30%2060.444))
vs
https://www.gbif.org/occurrence/search?has_coordinate=true&has_geospatial_issue=false&geometry=POLYGON%20((-30%2060.426,%20-30%2060.444,%2030%2060.444,%2030%2060.426,%20-30%2060.426))

**Comment by MortenHofft** [comment]
clearly visible here

![screen shot 2018-03-21 at 15 01 27](https://user-images.githubusercontent.com/6759267/37714020-c5e6efa2-2d18-11e8-87ac-828877fc056d.png)

https://www.gbif.org/occurrence/map?country=FI&has_coordinate=true&has_geospatial_issue=false&year=2006,2018&geometry=POLYGON%20((-180%2060.3,%20-180%2060.48,%20180%2060.48,%20180%2060.3,%20-180%2060.3))

**Comment by MortenHofft** [comment]
using decimalLatitude only they are different as expected
http://api.gbif.org/v1/occurrence/search?has_coordinate=true&has_geospatial_issue=false&decimalLatitude=60.444,60.462

http://api.gbif.org/v1/occurrence/search?has_coordinate=true&has_geospatial_issue=false&decimalLatitude=60.426,60.444

**Comment by MortenHofft** [comment]
same height but 3 different widths yielding different amount of outliers. That seem to support Tims initial idea that SOLR choose a resolution based on geom bounds

360 wide
![screen shot 2018-03-21 at 15 29 18](https://user-images.githubusercontent.com/6759267/37715624-cf086044-2d1c-11e8-9b93-8966463465d0.png)

60 wide
![screen shot 2018-03-21 at 15 29 22](https://user-images.githubusercontent.com/6759267/37715640-d8b2ef1a-2d1c-11e8-85c5-07452ff7ed06.png)

10 wide
![screen shot 2018-03-21 at 15 29 31](https://user-images.githubusercontent.com/6759267/37715670-e8c0bf90-2d1c-11e8-9977-d2fb2ec686a4.png)

**Comment by timrobertson100** [comment]
In our schema the [field `coordinate`](https://github.com/gbif/occurrence/blob/master/occurrence-search/solr/collection1/conf/schema.xml#L49) is defined as a `SpatialRecursivePrefixTreeFieldType` which I understand works along the lines of a GeoHash.

However, from [the docs](https://lucene.apache.org/solr/guide/6_6/spatial-search.html#SpatialSearch-RptWithGeometrySpatialField):

> The RptWithGeometrySpatialField field type is a derivative of SpatialRecursivePrefixTreeFieldType that also stores the original geometry internally in Lucene DocValues, which it uses to achieve accurate search.

This is also mentioned in [the SOLR 5.2 release notes](https://lucidworks.com/2015/06/09/whats-new-apache-solr-5-2/) 

My _suspicion_ is that SOLR estimates the prefix based on the geographic extent of the query hence the screenshots above.  My reading of the `RptWithGeometrySpatialField` suggests it would trim the results from the estimation to provide accurate search results.  

In short: I think we have the wrong type on the field [CC @fmendezh ?]

**Comment by timrobertson100** [comment]
I don't think so - a `point` is a `geometry` and the `RptWithGeometrySpatialField` stores the original value (as `docValue`) so that it can be consulted to provide accurate results. The results Morten observes (especially as he reduces the width of the window) and the docs suggest `SpatialRecursivePrefixTreeFieldType` will not yeild accurate results.

**Comment by timrobertson100** [comment]
We got to the bottom of this.  The links above are all using clockwise order geometries while the API/SOLR is configured to use counter clockwise and accept values that cross the dateline (e.g. tolerate 220 for -140).

When given correctly, we believe this works: 

https://www.gbif.org/occurrence/map?geometry=POLYGON((5.16375%2047.46495,293.73047%2047.46495,293.73047%2075.99739,5.16375%2075.99739,5.16375%2047.46495))&geometry=POLYGON((-186.54188%20-45.69047,-96.31687%20-45.69047,-96.31687%2076.62263,-186.54188%2076.62263,-186.54188%20-45.69047))
